changelog_unreleased/yaml/18324.md

```diff
@@ -0,0 +1,15 @@
+#### Use explicit key style for flow-mapping keys with trailing comments (#18324 by @kovsu, @fisker)
+
+<!-- prettier-ignore -->
+```yaml
+# Input
+{"foo": #comment
+bar}
```

Comment:
@kovsu The input is incorrect.

Reply:
Ha, so we also fixed this case. Added to tests [`c4d4975` (#18324)](https://github.com/prettier/prettier/pull/18324/commits/c4d4975697f364607c6c56d1285fea97f41e6ba0)